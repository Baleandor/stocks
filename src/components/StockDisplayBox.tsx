import type { StockSliceType } from "~/utils/types";

export default function StockDisplayBox({ stock }: { stock: StockSliceType }) {
  return (
    <div className="m-auto h-[155px] max-w-60 rounded-lg border border-green-400 p-2 text-left">
      <div>
        <span>Name: </span>
        <span>{stock.name}</span>
      </div>
      <div>
        <span>Symbol: </span>
        <span>{stock.symbol}</span>
      </div>
      <div>
        <span>Current Price: </span>
        <span>{stock.price}</span>
      </div>
      <div>
        <span>Daily Percentage Change: </span>
        <span>{stock.changePercent}%</span>
      </div>
    </div>
  );
}
