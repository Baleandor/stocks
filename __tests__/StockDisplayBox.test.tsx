import { render, screen } from "@testing-library/react";
import StockDisplayBox from "~/components/StockDisplayBox";
import type { StockSliceType } from "~/utils/types";


// Mock data for a stock
const mockStock: StockSliceType = {
  name: "Apple Inc.",
  symbol: "AAPL",
  price: 145.09,
  changePercent: 1.25,
};

describe("StockDisplayBox component", () => {
  it("renders the stock name", () => {
    render(<StockDisplayBox stock={mockStock} />); // ARRANGE

    const nameElement = screen.getByText("Apple Inc."); // ACT

    expect(nameElement).toBeInTheDocument(); // ASSERT
  });

  it("renders the stock symbol", () => {
    render(<StockDisplayBox stock={mockStock} />); // ARRANGE

    const symbolElement = screen.getByText("AAPL"); // ACT

    expect(symbolElement).toBeInTheDocument(); // ASSERT
  });

  it("renders the stock price", () => {
    render(<StockDisplayBox stock={mockStock} />); // ARRANGE

    const priceElement = screen.getByText("145.09"); // ACT

    expect(priceElement).toBeInTheDocument(); // ASSERT
  });

  it("renders the daily percentage change", () => {
    render(<StockDisplayBox stock={mockStock} />); // ARRANGE

    const changePercentElement = screen.getByText("1.25%"); // ACT

    expect(changePercentElement).toBeInTheDocument(); // ASSERT
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<StockDisplayBox stock={mockStock} />); // ARRANGE

    expect(asFragment()).toMatchSnapshot(); // ACT
  });
});
