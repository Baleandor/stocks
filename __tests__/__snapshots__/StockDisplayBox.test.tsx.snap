// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockDisplayBox component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="m-auto h-[155px] max-w-60 rounded-lg border border-green-400 p-2 text-left"
  >
    <div>
      <span>
        Name: 
      </span>
      <span>
        Apple Inc.
      </span>
    </div>
    <div>
      <span>
        Symbol: 
      </span>
      <span>
        AAPL
      </span>
    </div>
    <div>
      <span>
        Current Price: 
      </span>
      <span>
        145.09
      </span>
    </div>
    <div>
      <span>
        Daily Percentage Change: 
      </span>
      <span>
        1.25%
      </span>
    </div>
  </div>
</DocumentFragment>
`;
